src/js/components/modals/AppModalComponent.jsx

```diff
@@ -147,7 +147,7 @@ var AppModalComponent = React.createClass({
     event.preventDefault();
 
     if (!Object.keys(this.state.errorIndices).length) {
-      let app = AppFormStore.app;
+      let app = AppFormStore.getApp();
```

Comment:
well actually yes, this app needs never be mutated.  `AppsActions.applySettingsOnApp` makes its own local copy of the `app` (see `src/js/actions/AppsActions.js:147`), and as a result of a successful `AppsActions.createApp` the AppStore makes yet another local copy for processing it (see `src/js/stores/AppsStore.js:121`). What do you think?